modules/sequence/src/main/scala/lucuma/odb/sequence/data/OffsetGenerator.scala

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.sequence.data
+
+import cats.Eq
+import cats.data.NonEmptyList
+import cats.derived.*
+import cats.syntax.eq.*
+import cats.syntax.option.*
+import lucuma.core.math.Angle
+import lucuma.core.math.Offset
+import lucuma.core.model.sequence.TelescopeConfig
+import lucuma.odb.sequence.syntax.hash.*
+import lucuma.odb.sequence.util.HashBytes
+import monocle.Lens
+import monocle.Optional
+import monocle.Prism
+import monocle.macros.GenLens
+import monocle.macros.GenPrism
+
+sealed trait OffsetGenerator:
+
+  import OffsetGenerator.*
+
+  def offsetGeneratorType: OffsetGeneratorType =
+    this match
+      case NoGenerator   => OffsetGeneratorType.NoGenerator
+      case Enumerated(_) => OffsetGeneratorType.Enumerated
+      case Random(_, _)  => OffsetGeneratorType.Random
+      case Spiral(_, _)  => OffsetGeneratorType.Spiral
+      case Uniform(_, _) => OffsetGeneratorType.Uniform
+
+object OffsetGenerator:
+
+  case object NoGenerator extends OffsetGenerator
+
+  case class Enumerated(
```

Comment:
This may clash with core's `Enumerated`. is this manual? or user supplied?

Reply:
The idea with enumerated offsets is that the user explicitly supplies a list.  This list is conceptually repeated infinitely and we pick the first _n_ values from it.